packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -557,28 +537,61 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin
 /// node, which is annotated with the scrolling actions, will house the
 /// scrollable children.
 class _ExcludableScrollSemantics extends SingleChildRenderObjectWidget {
-  const _ExcludableScrollSemantics({ Key key, Widget child }) : super(key: key, child: child);
+  const _ExcludableScrollSemantics({
+    Key key,
+    @required this.position,
+    Widget child
+  }) : assert(position != null), super(key: key, child: child);
+
+  final ScrollPosition position;
+
+  @override
+  _RenderExcludableScrollSemantics createRenderObject(BuildContext context) => new _RenderExcludableScrollSemantics(position: position);
 
   @override
-  _RenderExcludableScrollSemantics createRenderObject(BuildContext context) => new _RenderExcludableScrollSemantics();
+  void updateRenderObject(BuildContext context, _RenderExcludableScrollSemantics renderObject) {
+    renderObject.position = position;
+  }
 }
 
 class _RenderExcludableScrollSemantics extends RenderProxyBox {
-  _RenderExcludableScrollSemantics({ RenderBox child }) : super(child);
+  _RenderExcludableScrollSemantics({
+    @required ScrollPosition position,
+    RenderBox child,
+  }) : _position = position, assert(position != null), super(child) {
+    position.addListener(markNeedsSemanticsUpdate);
+  }
+
+  /// Whether this render object is excluded from the semantic tree.
+  ScrollPosition get position => _position;
+  ScrollPosition _position;
+  set position(ScrollPosition value) {
+    assert(value != null);
+    if (value == _position)
+      return;
+    _position.removeListener(markNeedsSemanticsUpdate);
+    _position = value;
+    _position.addListener(markNeedsSemanticsUpdate);
+    markNeedsSemanticsUpdate();
+  }
+
+  double get _scrollProgress {
+    return (position.pixels - position.minScrollExtent) / (position.maxScrollExtent - position.minScrollExtent);
```

Comment:
`minScrollExtent` can be `double.negativeInfinity` and `maxScrollExtent` can be `double.infinity`, so I don't think this does what you want it to.